addon/commands/besluit-plugin/insert-article.js

```diff
@@ -0,0 +1,63 @@
+import { v4 as uuid } from 'uuid';
+import { insertHtml } from '@lblod/ember-rdfa-editor/commands/insert-html-command';
+
+export default function InsertArticleCommand(
+  controller,
+  articleContent,
+  articleNumber
+) {
+  let range = controller.state.selection;
+  controller.state.doc.descendants((node, pos) => {
+    const typeOfAttribute = node.attrs['typeof'];
+    if (
+      typeOfAttribute?.includes('besluit:Besluit') ||
+      typeOfAttribute?.includes('http://data.vlaanderen.be/ns/besluit#Besluit')
+    ) {
+      range = { from: pos + node.nodeSize - 2, to: pos + node.nodeSize - 2 };
+      return false;
+    }
+  });
```

Comment:
In order to determine the current Besluit, it is preferred to use the datastore (you can use the limitToRange method).